ui/ecsconfig.py

```diff
@@ -626,61 +665,145 @@ def add_all():
 
     if a:
         n = None
-        results = add_all()
-        for result in results:
-            o('Created replication group {}'.format(result['name']))
+        available_rg_configs = list_all()
+        if available_rg_configs is not None:
+            results = add_all()
+            for result in results:
+                o('Created replication group {}'.format(result['name']))
+        else:
+            o('No replication group configurations in deploy.yml')
 
     if n is not None:
         result = add_rg(n)
         o('Created replication group {}'.format(result['name']))
 
 
-@ecsconfig.command('namespace', short_help='Work with ECS Namespaces')
-@click.option('-l', is_flag=True, help='List known namespace configs')
-@click.option('-r', is_flag=True, help='Get current namespace configs from ECS')
-@click.option('-a', is_flag=True, help="Add all namespaces to ECS")
-@click.option('-n', default=None, help='Add the given namespace to ECS')
-@pass_conf
-def namespace(conf, l, r, a, n):
-    def list_all():
-        return conf.ecs.get_ns_names()
-
-    def get_all():
-        return conf.api_client.namespace.list()
-
-    def add_namespace(namespace_name):
-        o('Adding namespace {}'.format(namespace_name))
-        ns_dict = conf.ecs.get_ns_options(namespace_name)
-
-        kwargs = {"is_stale_allowed": ns_dict['is_stale_allowed'],
-                  "is_compliance_enabled": ns_dict['is_compliance_enabled'],
-                  "is_encryption_enabled": ns_dict['is_encryption_enabled'],
-                  "namespace_admins": ns_dict['namespace_admins'],
-                  "default_data_services_vpool": ns_dict['']}
-
-
-        return conf.api_client.namespace.create(namespace_name, **kwargs)
-
-    def add_all():
-        for namespace_name in list_all():
-            add_namespace(namespace_name)
-
-    if l:
-        o('Available Namespace configurations:')
-        for ns_name in list_all():
-            o('\t{}'.format(ns_name))
-    if r:
-        o('Namespaces currently configured:')
-        namespaces = get_all()
-        for ns_data in namespaces['namespace']:
-            o('\t{}'.format(ns_data['name']))
-    if a:
-        n = None
-        add_all()
-    if n is not None:
-        add_namespace(n)
+# @ecsconfig.command('namespace', short_help='Work with ECS Namespaces')
+# @click.option('-l', is_flag=True, help='List known namespace configs')
+# @click.option('-r', is_flag=True, help='Get current namespace configs from ECS')
+# @click.option('-a', is_flag=True, help="Add all namespaces to ECS")
+# @click.option('-n', default=None, help='Add the given namespace to ECS')
+# @pass_conf
+# def namespace(conf, l, r, a, n):
+#     """
+#     # BUG: Broken - doesn't build NS right
```

Comment:
Not sure, haven't spent time on it.  You're more than welcome to own #221 if you want to dig into it.